sky/data/storage.py

```diff
@@ -420,9 +431,11 @@ def __init__(self,
             # syncing to file_mount stage..
             if self.sync_on_reconstruction:
```

Comment:
I'm holding off on this for now because this would also affect the class constructor and thus the external interface. The eventual plan is to remove this arg from the Storage ctor, after which we should rename it.